File: 2540.getCommon.go
package main

func getCommon(numsa []int, numsb []int) int {
	ptra := 0
	ptrb := 0

	for ptra < len(numsa) && ptrb < len(numsb) {
		if numsa[ptra] == numsb[ptrb] {
			return numsa[ptra]
		}

		if numsa[ptra] > numsb[ptrb] {
			ptrb++
		} else {
			ptra++
		}
	}

	return -1
}

File: 2540.getCommon.c
#include "shdstd.h"

int getCommon(int *numsa, int a_size, int *numsb, int b_size)
{
	int ptra = 0;
	int ptrb = 0;

	while (ptra < a_size && ptrb < b_size) {
		if (numsa[ptra] == numsb[ptrb]) {
			return numsa[ptra];
		}

		if (numsa[ptra] > numsb[ptrb]) {
			++ptrb;
		} else {
			++ptra;
		}
	}

	return -1;
}
